spyder/utils/syntaxhighlighters.py

```diff
@@ -478,7 +479,12 @@ def highlight_block(self, text):
                                 self.found_cell_separators = True
                                 oedata = OutlineExplorerData()
                                 oedata.text = to_text_string(text).strip()
-                                oedata.fold_level = start
+                                if text.lstrip().startswith(("# %%%", "#%%%")):

```

Comment:
You should be able to allow for an unlimited number of levels and make your code cleaner to boot if you just count the number of ``"%"``s after the initial ``# %%`` or ``#%%`` and set your offset/fold level to ``base_value`` + ``count_%`` * 4.

Reply:
I agree.  The code could be cleaner here.  I'll see what I need to do to fix "cell blocks in code blocks" and then clean it up.  Even the `4` could get messy with those "2-space" people ;-)

After looking at the "cell blocks in code blocks" issue, I think more than one subgroup level would be a testing nightmare with lots of corner cases.  What would happen if some one decorates the code block with `# %%%%%%%%%%% Cell Name %%%%%%%%%%%%`? This currently makes a code block, but you can see the problem.